milli/src/index.rs

```diff
@@ -857,6 +864,56 @@ impl Index {
     ) -> heed::Result<()> {
         self.main.put::<_, Str, SerdeJson<DateTime<Utc>>>(wtxn, main_key::UPDATED_AT_KEY, &time)
     }
+
+    // user management
+    pub fn user_add_document_filter(
+        &self,
+        wtxn: &mut RwTxn,
+        user: &str,
+        mut docids: Vec<String>,
+    ) -> Result<()> {
+        docids.sort_unstable();
+        let mut external_documents_ids = self.external_documents_ids(wtxn)?;
+        let mut new_docids = fst::MapBuilder::memory();
+        let mut bitmap = RoaringBitmap::new();
+        let documents_ids = self.documents_ids(wtxn)?;
+        let mut available_documents_ids = AvailableDocumentsIds::from_documents_ids(&documents_ids);
+        for id in docids {
+            match external_documents_ids.get(&id) {
+                Some(id) => {
+                    bitmap.insert(id);
+                }
+                None => {
+                    let next = available_documents_ids.next().unwrap();
+                    bitmap.insert(next);
+                    new_docids.insert(id, next as u64)?;
+                }
+            }
+        }
```

Comment:
this doesn't work if the filter is added before any document, need to figure this out